### ISSUE TITLE
Support Razor Class Libraries

### ISSUE THREAD

**Issue by TheJayMann** [giraffe-fsharp/Giraffe.Razor#6]
I tried creating a solution containing a giraffe web application and a razor class library, adding a reference from the class library to the webapp.  After setting up the webapp to be able to use the razor view engine and creating a razorHtmlView result, and adding the corresponding view to the class library, I received and error when attempting to load the page stating that the Views folder does not exist, even though the folder does indeed exist in the class library.

As I was typing this, I decided to add an empty folder called Views to my webapp project just to see what happens.  By my surprise, the view from the class library was indeed loaded, and everything works as I would like it to work.  However, it still bugs me that I have to have an empty folder in order to use the class library.  If possible, I would like to request something that will allow the application to tell the razor view engine to work even if the folder doesn't exist on disk, such as possibly an optional parameter isRazorLibrary, or an extension method similar to AddRazorLibraryEngine, so that it can work seamlessly with razor class libraries.

**Comment by dustinmoris** [comment]
Hi,

I agree that it's not clean to have to create an empty folder for it to work. I'll have a dig at this over the weekend or if you want you can also provide a PR if you know what needs to be done. I haven't used razor views from a class library yet, so it will probably take me a bit of playing around to fully understand what's happening and how to fix it properly.

**Comment by TheJayMann** [comment]
I took a look at what the code is doing now.  It is simply creating a PhysicalFileProvider pointed at the specified directory, while also clearing out all other previously defined FileProviders.  I don't know too much about razor class libraries just yet, but this confirms that they are able to be accessed when only a PhysicalFileProvider exists and the requested view does not physically exist.  At this point my guess is that the existence of a razor class library either adds an additional FileProvider (in which case there is no need to call AddRazorEngine at all), or that PhysicalFileProvider (and possibly other built in FileProviders) will attempt to load a razor view from a razor class library if it was unable to actually find a file.  I'll look more into this, and, if I find something before you do, I'll either update this issue or submit a PR.

**Comment by dustinmoris** [comment]
Awesome, thanks for looking into it. Sounds like you're on the right path!
I'll let you know about my progress over the weekend as well. I think this
is going to be a couple of hours research with a small fix in the end :).

**Comment by TheJayMann** [comment]
So, this may be simpler than I thought.  The AddRazorEngine method does three things.  First is that it clears out all FileProviders and provides it's own based on the path given to it.  Second is that it adds mvc.  Finally, it adds antiforgery.  I created a new project where I did not use AddRazorEngine and instead only did AddMvc.  I do not have a Views folder in my web app project, and I do have a Views folder in my RCL project with an Index.cshtml.  I call razorHtmlView with a view name of "Index", and everything works.  At this point, I would think the only thing that needs to be changed is the documentation, or possibly an overload of AddRazorEngine accepting no parameters which simply calls AddMvc().AddAntiforgery().
